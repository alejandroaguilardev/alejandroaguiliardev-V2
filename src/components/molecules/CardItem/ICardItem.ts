export interface ICardItem {
	title: string;
	subtitle: string;
	description: string;
	icon: string;
	size?:string;
}
