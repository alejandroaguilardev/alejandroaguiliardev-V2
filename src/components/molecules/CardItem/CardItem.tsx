import { Card } from "@mui/material"
import { CardContent } from "../../atoms"
import { ICardItem } from "./ICardItem"

export const CardItem = ({ title, subtitle, description, icon, size = 'small' }: ICardItem) => {
    return (
        <Card
            sx={{
                alignItems: 'center',
                backgroundColor: 'background.paper',
                boxShadow: '1px 1px 30px #ddd',
                cursor: 'pointer',
                display: 'flex',
                flexDirection: 'column',
                justifyContent: 'center',
                height: `${size === "small" ? "15rem" : "22rem"}`,
                py: 2,
                px: 2,
                transition: '.7s  all ',
                "&:hover": {
                    backgroundColor: `${size === 'small' ? 'primary.main' : 'background.paper'}`,
                    color: `${size === 'small' ? 'primary.contrastText' : 'inherit'}`,
                    transform: 'translate(0,-5px)'
                }
            }}>
            <CardContent
                title={title}
                subtitle={subtitle}
                description={description}
                icon={icon}
                size={size}
            />
        </Card>
    )
}
