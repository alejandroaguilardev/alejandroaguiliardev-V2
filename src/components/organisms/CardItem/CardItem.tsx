import { Card } from "@mui/material";
import { useThemeContext } from "../../../context";
import { CardContent } from "../../molecules";
import { ICardItem } from "./ICardItem";

export const CardItem = ({
    title,
    subtitle,
    description,
    icon,
    size = 'small',
    link = "",
    repository = ""
}: ICardItem) => {
    const { theme } = useThemeContext();

    return (
        <Card
            sx={{
                alignItems: "center",
                backgroundColor: 'background.paper',
                boxShadow: theme.mode === 'dark' ? '' : '1px 1px 30px #ddd',
                cursor: 'pointer',
                display: 'flex',
                flexDirection: 'column',
                justifyContent: size === "large" ? "start" : "center",
                height: size === "small" ? "15rem" : "23rem",
                p: 0,
                m: 0,
                transition: '.7s  all',
                "&:hover": {
                    backgroundColor: size === 'small' ? 'primary.main' : 'background.paper',
                    color: size === 'small' ? 'primary.contrastText' : 'inherit',
                    transform: 'translate(0,-5px)'
                }
            }}>
            <CardContent
                title={title}
                subtitle={subtitle}
                description={description}
                icon={icon}
                size={size}
                link={link}
                repository={repository}
            />
        </Card>
    )
}
