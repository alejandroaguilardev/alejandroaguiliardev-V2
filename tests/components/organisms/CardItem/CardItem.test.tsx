import { render, screen } from "@testing-library/react";
import { CardItem } from "../../../../src/components/organisms/CardItem/CardItem";

describe("CardItem Test", () => {
	const title = "title";
	const subtitle = "subtitle";
	const description = "titdescriptione";
	const icon = "/img/banner";

	test("toMacthSnapshot", () => {
		const { container } = render(
			<CardItem
				title={title}
				subtitle={subtitle}
				description={description}
				icon={icon}
			/>
		);

		expect(container).toMatchSnapshot();
		expect(screen.getByText(title)).toBeTruthy();
		expect(screen.getByText(subtitle)).toBeTruthy();
		expect(screen.getByText(description)).toBeTruthy();
	});
});
