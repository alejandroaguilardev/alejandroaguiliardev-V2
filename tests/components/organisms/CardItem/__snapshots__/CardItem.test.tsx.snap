// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardItem Test toMacthSnapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-15cnh1e-MuiPaper-root-MuiCard-root"
  >
    <div
      class="MuiBox-root css-18i215d"
    >
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2770%27%20height=%2770%27/%3e"
            style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
          />
        </span>
        <img
          alt="title"
          data-nimg="intrinsic"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
    </div>
    <div
      class="MuiBox-root css-gajhq5"
    >
      <h3
        class="MuiTypography-root MuiTypography-h3 MuiTypography-alignCenter css-yg6atj-MuiTypography-root"
      >
        title
      </h3>
      <p
        class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter css-1a3kej9-MuiTypography-root"
      >
        subtitle
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter css-r5yzcc-MuiTypography-root"
      >
        titdescriptione
      </p>
    </div>
  </div>
</div>
`;
